src/AzureExtension/Providers/RepositoryProvider.cs

```diff
@@ -394,4 +346,315 @@ public void Dispose()
     {
         GC.SuppressFinalize(this);
     }
+
+#pragma warning disable CA1309 // Use ordinal string comparison. I want to use linguistic comparison.
+    public IAsyncOperation<RepositoriesResult> GetRepositoriesAsync(IReadOnlyDictionary<string, string> fieldValues, IDeveloperId developerId)
+    {
+        // First query will find repos only if the user made a PR.
+        var shouldIgnorePRDateOld = _shouldIgnorePRDate;
+        _shouldIgnorePRDate = true;
+        return Task.Run(() =>
+        {
+            try
+            {
+                var serverToUse = fieldValues.ContainsKey(_server) ? fieldValues[_server] : string.Empty;
+                var organizationToUse = fieldValues.ContainsKey(_organization) ? fieldValues[_organization] : string.Empty;
+                var projectToUse = fieldValues.ContainsKey(_project) ? fieldValues[_project] : string.Empty;
+
+                // Get access token for ADO API calls.
+                if (developerId is not DeveloperId.DeveloperId azureDeveloperId)
+                {
+                    var exception = new NotSupportedException("The DeveloperId is not valid.");
+                    return new RepositoriesResult(exception, $"{exception.Message} HResult: {exception.HResult}");
+                }
+
+                _azureHierarchy ??= new AzureRepositoryHierarchy(azureDeveloperId);
+
+                var organizations = _azureHierarchy.GetOrganizations();
+                var orgsInModernUrlFormat = new List<Organization>();
+                if (serverToUse.Equals(_defaultSearchServerName))
+                {
+                    // For a default search, look up all orgs with the modern URL format
+                    orgsInModernUrlFormat = organizations.Where(x => x.AccountUri.Host.Equals(_defaultSearchServerName)).ToList();
+                }
+
+                // If the user is apart of orgs in the modern URL format, use these going forward.
+                if (orgsInModernUrlFormat.Count > 0)
+                {
+                    organizations = orgsInModernUrlFormat;
+                }
+
+                if (!string.IsNullOrEmpty(organizationToUse))
+                {
+                    organizations = organizations.Where(x => x.AccountName.Equals(organizationToUse)).ToList();
+                }
+
+                var options = new ParallelOptions()
+                {
+                    // Let the program decide
+                    MaxDegreeOfParallelism = -1,

```

Comment:
I think if you don't add MaxDegreeOfParallelism, thats the default behavior. But up to you if you want to change it.

Reply:
Removed.  I thought I needed the options argument.